fdb-relational-core/src/test/java/com/apple/foundationdb/relational/recordlayer/query/GroupByQueryTests.java

```diff
@@ -46,6 +49,120 @@ public GroupByQueryTests() {
         Utils.enableCascadesDebugger();
     }
 
+    @Disabled // doesn't work for serializationMode = TO_OLD, re-enable after serializationMode = TO_NEW
+    @Test
+    void groupByWithScanLimit() throws Exception {
+        final String schemaTemplate =
+                "CREATE TABLE T1(pk bigint, a bigint, b bigint, c bigint, PRIMARY KEY(pk))" +
+                        "CREATE INDEX idx1 as select a, b, c from t1 order by a, b, c";
+        try (var ddl = Ddl.builder().database(URI.create("/TEST/QT")).relationalExtension(relationalExtension).schemaTemplate(schemaTemplate).build()) {
+            try (var conn = ddl.setSchemaAndGetConnection()) {
+                conn.setOption(Options.Name.EXECUTION_SCANNED_ROWS_LIMIT, 2);
+                try (var statement = conn.createStatement()) {
+                    insertT1Record(statement, 2, 1, 1, 20);
+                    insertT1Record(statement, 3, 1, 2, 5);
+                    insertT1Record(statement, 4, 1, 2, 15);
+                    insertT1Record(statement, 5, 1, 2, 5);
+                    insertT1Record(statement, 6, 2, 1, 10);
+                    insertT1Record(statement, 7, 2, 1, 40);
+                    insertT1Record(statement, 8, 2, 1, 20);
+                    insertT1Record(statement, 9, 2, 1, 90);
+
+                    String query = "SELECT a AS OK, b, MAX(c) FROM T1 GROUP BY a, b";
+                    Continuation continuation = null;
+                    // scan pk = 2 and pk = 3 and hit SCAN_LIMIT_REACHED
+                    Assertions.assertTrue(statement.execute(query), "Did not return a result set from a select statement!");
+                    try (final RelationalResultSet resultSet = statement.getResultSet()) {
+                        ResultSetAssert.assertThat(resultSet).hasNextRow()
+                                .isRowExactly(1L, 1L, 20L)
+                                .hasNoNextRow();
+                        continuation = resultSet.getContinuation();
+                    }
+                    // scan pk = 5 and pk = 4 rows, hit SCAN_LIMIT_REACHED
+                    String postfix = "EXECUTE CONTINUATION " + Base64.getEncoder().encodeToString(continuation.serialize());
```

Comment:
This doesn't look right to me. `EXECUTE CONTINUATION` should be its own command rather than a post fix to the query

Reply:
modified.